pydrofoil/real.py

```diff
@@ -0,0 +1,99 @@
+import sys
+from rpython.rlib.rbigint import rbigint, _divrem as bigint_divrem
+from rpython.rlib.rarithmetic import r_uint, intmask, string_to_int, ovfcheck, \
+        int_c_div, int_c_mod, r_ulonglong
+
+class Real(object):
+    def __init__(self, num, den):
+        self.num = num
+        self.den = den
+
+    @staticmethod
+    def fromint(num, den=1):
+        sign = num/abs(num)*den/abs(den)
+        num = rbigint.fromint(abs(num))
+        den = rbigint.fromint(abs(den)*sign)
```

Comment:
I don't know whether there is a convention about this in general, but I would prefer to put the sign into the numerator, not the denominator. The denominator already has a limitation, it cannot be `0`. so making that stronger and saying that it can't be negative either feels sensible.

note also that you have a bug here: if you `num` or `den` are MININT (`= -2**63`) then `abs(num)` is undefined. so you need to check for that value explicitly and add unit tests for it.

Reply:
ok sure, I would change that, I put the sign into the numerator is just because I can use the same equation for ceil or floor based on divmod method of rbigint, but I could change it easily. I would also check the corner test case. Thanks for refering.